src/js/providers/data-normalizer.ts

```diff
@@ -7,7 +7,9 @@ export type QualityLevel = {
     height?: number;
 };
 
-export function qualityLevel(level: QualityLevel & GenericObject): QualityLevel {
+export type Level = QualityLevel & GenericObject;
```

Comment:
I think it was originally defined this way because of the types of levels you mentioned below (hls, shaka, etc.) and how they have many additional properties.

I'll take another look at where bandwidth is defined.

Reply:
Outside of bandwidthEstimate in the config, from what I can tell, `bandwidth` is only really defined on the provider levels that use it so like Shaka levels, Shaka Audio levels and flash levels (I'm assuming). We never define it.